semtypes/src/main/java/io/ballerina/types/definition/ListDefinition.java

```diff
@@ -98,6 +127,35 @@ public ComplexSemType define(Env env, List<SemType> members, SemType rest) {
         return this.createSemType(env, ro, rw);
     }
 
+    private ListAtomicType readOnlyListAtomicType(ListAtomicType ty) {
+        if (Common.typeListIsReadOnly(ty.members.initial)
+                && Core.isReadOnly(ty.rest)) {
+            return ty;
+        }
+        return ListAtomicType.from(
+                FixedLengthArray.from(
+                        List.of(Common.readOnlyTypeList(ty.members.initial.toArray(new SemType[0]))),
```

Comment:
@rdhananjaya should we overload `readOnlyTypeList` with an Iterable parameter too?

Reply:
+1